src/ops/direnv/mod.rs

```diff
@@ -38,48 +39,54 @@ pub fn main(project: Project) -> OpResult {
 
         // Ping sent & paths aren't cached: once the environment is created
         // the direnv environment will be updated automatically.
-        (true, false) => {
-            eprintln!("Notice: lorri has not completed an evaluation for this project yet.");
-            eprintln!("        lorri should be evaluating the environment now.");
-        }
+        (true, false) =>
+            info!(
+                "lorri has not completed an evaluation for this project yet"
+            ),
 
         // Ping not sent and paths are cached: we can load a stale environment
         // When the daemon is started, we'll send a fresh ping.
-        (false, true) => {
-            eprintln!("Info: the lorri daemon is not running. Loading a cached environment.");
-        }
+        (false, true) =>
+            info!(
+                "lorri daemon is not running, loading a cached environment"
+            ),
 
         // Ping not sent and paths are not cached: we can't load anything,
         // but when the daemon in started we'll send a ping and eventually
         // load a fresh environment.
-        (false, false) => {
-            eprintln!("Error: the lorri daemon is not running and this project has not yet been evaluated.");
-            eprintln!("       Please run `lorri daemon`.");
-        }
+        (false, false) =>
+            error!("lorri daemon is not running and this project has not yet been evaluated, please run `lorri daemon`"),
     }
 
     if std::env::var("DIRENV_IN_ENVRC") != Ok(String::from("1")) {
-        eprintln!(
-            "Warning: 'lorri direnv' should be executed by direnv from within an `.envrc` file."
-        )
+        warn!("`lorri direnv` should be executed by direnv from within an `.envrc` file")
     }
 
-    ok_msg(format!(
+    // direnv interprets stdout as a script that it evaluates. That is why (1) the logger for
```

Comment:
I'm not sure about this change. I quite liked that the caller dealt with what to do with the result, instead of having to pass in different writers. If the caller is already capable of passing in the correct writer, it seems the caller is already capable of dealing with the return value, no?

Reply:
This way the caller still decides what to do with the output, and arguably it's more explicit (upside in my book) but also more verbose (downside). The way the diff currently also just makes it as unclear as previously what should actually be done with the result. I'll have another think about this and perhaps try to make it closer to what it was before.

Reply:
I've pushed a different proposal that changes `OpResult` to be `Result<(), ExitError>` after noticing that `ok_msg` was hardly used. I think this makes the code easier to read.